lib/friendly_shipping/services/usps_ship/serialize_rate_estimates_request.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+require 'friendly_shipping/services/usps/machinable_package'
+
+module FriendlyShipping
+  module Services
+    class USPSShip
+      class SerializeRateEstimatesRequest
+        class << self
+          # Serialize a rate estimates request.
+          #
+          # @param shipment [Physical::Shipment]
+          # @param options [RateEstimateOptions]
+          # @return [Hash]
+          def call(shipment:, options:)
+            length, width, height = dimensions(shipment)
+            {
+              originZIPCode: shipment.origin.zip,
+              destinationZIPCode: shipment.destination.zip,
+              weight: shipment.packages.sum(&:weight).convert_to(:lbs).value.to_f.round(2),
+              length: length.convert_to(:in).value.to_f.round(2),
+              width: width.convert_to(:in).value.to_f.round(2),
+              height: height.convert_to(:in).value.to_f.round(2),
+              mailClass: options.shipping_method.service_code,
+              processingCategory: options.processing_category,
+              rateIndicator: options.rate_indicator,
+              destinationEntryFacilityType: options.destination_entry_facility_type,
+              priceType: options.price_type,
+              mailingDate: options.mailing_date.strftime("%Y-%m-%d")
+            }
+          end
+
+          private
+
+          # @param shipment [Physical::Shipment]
+          def dimensions(shipment)
+            length = shipment.packages.sum(&:length)
+            width = shipment.packages.sum(&:width)
+            height = shipment.packages.sum(&:height)
+            [length, width, height].sort.reverse
+          end
```

Comment:
Won't this be too large? Consider three packages that are all 4x3x2. If they are stacked the shipment would be 4x3x6, not 12x9x6. Maybe this is outside the scope of this PR but thought I would bring it up.

Reply:
Good catch. I updated to add the lengths and use the max width and height. Best we can do for now since the new API does not support passing multiple packages in a single call (unlike the legacy API).